Clarify the overall structure of a naming rule (#21703)
dotnet/docs: docs/fundamentals/code-analysis/style-rules/naming-rules.md

---
# Naming rules

Naming rules let you specify and enforce the naming of .NET programming language code elements, such as classes, properties, and methods. For example, you can specify that public members must be capitalized or that private fields must begin with `_`.

A naming rule has three parts:

> * The _symbol group_ it applies to. In our examples above, that's the "public members" or "private fields" portion of the rule.

Reviewer: On this line and line 27, I'm not sure what the "examples above" refer to.

Author: The example on line 22. Would it help to maybe make those examples a separate list so they stand out?

> * The _naming style_ to associate with the rule. In our examples above, that's the "must be capitalized" or "must begin with `_`" portion of the rule.

Reviewer: ```suggestion
* The _naming style_ to associate with the rule, for example, that the name must be capitalized or start with an underscore.
```

Author: Ah, that works well, great suggestion.

* The severity for enforcing the convention.

> You define naming rules in an EditorConfig file. First, you must define the symbol group and naming style, and give each of them a title. Then you specify the naming rule which links everything together.

Reviewer: ```suggestion
You define naming rules in an EditorConfig file. First, you must specify the symbol group and naming style and give each of them a title. Then you specify the naming rule, which links everything together.
```
